gemini/use-cases/education/resume_generator.ipynb

```diff
@@ -0,0 +1,877 @@
+      },
+      "outputs": [],
+      "source": [
+        "from ast import literal_eval\n",
+        "\n",
+        "# Native Imports\n",
+        "import os\n",
+        "\n",
+        "# Third-party Imports\n",
+        "import docx\n",
+        "import docx2txt\n",
+        "from docx.document import Document\n",
+        "from docx.oxml.table import CT_Tbl\n",
+        "from docx.oxml.text.paragraph import CT_P\n",
+        "from docx.table import Table, _Cell\n",
+        "from docx.text.paragraph import Paragraph\n",
+        "import pypdf\n",
+        "from vertexai.generative_models import GenerativeModel"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "5303c05f7aa6"
+      },
+      "source": [
+        "### Import a sample resume file"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "5c4978c07d9a"
+      },
+      "outputs": [],
+      "source": [
+        "! gsutil cp gs://github-repo/generative-ai/gemini/use-cases/education/resume-generation/Sample_1.docx ."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "e43229f3ad4f"
+      },
+      "source": [
+        "### Load model"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "cf93d5f0ce00"
+      },
+      "outputs": [],
+      "source": [
+        "MODEL_NAME = \"gemini-1.5-flash\"  # Choice of model\n",
+        "generation_config = {\"temperature\": 0}\n",
+        "\n",
+        "model = GenerativeModel(\n",
+        "    model_name=MODEL_NAME,\n",
+        "    generation_config=generation_config,\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "VPylX87eREVh"
+      },
+      "source": [
+        "## Notebook Tutorial"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "oP_UflnpUVRL"
+      },
+      "source": [
+        "### Resume Structure and Prompt Initialization\n",
+        "\n",
+        "After installing necessary packages and loading the language model, we initialize predefined prompts designed for resume generation and the related structure. These prompts guide the model to extract relevant information from an existing resume and create a tailored output that aligns with the provided job description."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "1d17bfb0-f4ee-46b9-87b5-7b04ca6a1129"
+      },
+      "outputs": [],
+      "source": [
+        "\"\"\"\n",
+        "------------------------\n",
+        "EXTRACTION\n",
+        "------------------------\n",
+        "Personal Information\n",
+        "Education\n",
+        "Languages\n",
+        "Extracurricular_Activities\n",
+        "References\n",
+        "Experience\n",
+        "\n",
```

Comment:
These values are duplicated in multiple prompts and variables. Can you find a way to have a single source of truth for this? Load a variable into the prompt instead of hard coding.

Reply:
Removed the above and used single reference